frontend/src/lib/SegmentMetricsManager.ts

```diff
@@ -211,7 +212,23 @@ export class SegmentMetricsManager implements MetricsManager {
 
   // eslint-disable-next-line class-methods-use-this
   private track(evName: string, data: Record<string, unknown>): void {
-    analytics.track(evName, data)
+    let parser = new UAParser("user-agent")
+    let parsed_ua = parser.getResult()
+
+    analytics.track(evName, data, {
+      context: {
+        ip: "0.0.0.0",
+        page: {
+          title: "",
+        },
+        browser: parsed_ua.browser.name,
+        browser_version: parsed_ua.browser.version,
+        os: parsed_ua.os.name,
+        os_version: parsed_ua.os.version,
+        device_type: parsed_ua.device.type,
+        userAgent: "",
+      },
+    })
```

Comment:
This looks great! Needs a couple things before it gets merged:

1. Can you add a comment in the function body explaining why we're zeroing out the IP and userAgent, to make it clear that their absence isn't just an oversight. ("We zero out the IP and userAgent data because we don't want to collect them.")
2. Can you add a test to `SegementMetricsManager.test.ts` that asserts the analytics.track function is called as expected, so we don't have regressions?

Reply:
Will do!

Reply:
@tconkling I tried to add tests -- let me know if they look reasonable